psl-core/src/main/java/org/linqs/psl/application/inference/InferenceApplication.java

```diff
@@ -74,7 +77,9 @@
     protected GroundRuleStore groundRuleStore;
     protected TermStore termStore;
     protected TermGenerator termGenerator;
-    protected AtomManager atomManager;
+    protected PersistedAtomManager atomManager;
```

Comment:
Almost all the IA's construct the PAM in the same way (except lazy).
Let's factor it out to a method that lazy can override.
Something like `initializeAtomManager()`.